### DISCUSSION TITLE
Complex prompts / BREAK in regions

### DISCUSSION THREAD

**Discussion by siilky** [hako-mikan/sd-webui-regional-prompter#397] [Ideas]
Hi, can the code be updated to allow BREAKing inside regions and long prompts for regions in matrix mode? I'm not sure yet what breaks the prompt with long prompts in regions, but BREAK handling is obviously broken with using explicit ADDx keywords. Also keywording can be improved, like strict order of COMM/BASE is not obvious in use and could be relaxed.
I could help with coding, and have already looked at the code (regioning flow is quite questionable but I could handle that), but I'm not sure about the prompting details over regions and would need advice about that.

**Comment by Heather95** [comment]
@siilky Nobody answered you, so I'll recommend trying mask mode or prompt mode. Both of these allow for finer control over region sizes and thus splitting your prompt into more parts than possible with matrix mode.

**Comment by hako-mikan** [comment]
Is the same issue to https://github.com/hako-mikan/sd-webui-regional-prompter/issues/286 ?
If so, new option added.

**Comment by siilky** [comment]
there are several problems which mostly relate to changing boundaries within schedule/batch:
- plugin parses only the first prompt in batch and if regions are changing over batch it is rendered incorrectly,
- in the same way, when boundary change happens within single prompt schedule it is rendered incorrectly also. 
Along with some improvements that could be added:
- strict order of addbase/addcomm might be relaxed 
- adding sort of REGION keyword instead of BREAK to avoid interfering with it, also unbinding prompt from regions at some degree, to move forward to visual region setting.
I have implemented some code already for that but stopped at the point where scheduling occurs inside the app, and it is not easily customizable/interceptable without hacks.
